sharktank/tests/models/llama/benchmark-tests.py

```diff
@@ -0,0 +1,310 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import logging
+import os
+import unittest
+import pytest
+import subprocess
+from pathlib import Path
+from typing import List
+
+
+class BaseBenchmarkTest(unittest.TestCase):
+    def setUp(self):
+        raise NotImplementedError("Subclasses should implement this method.")
+
+    def get_export_cmd(
+        self,
+        attention_kernel: str,
+        irpa_path: str,
+        output_mlir_path: str,
+        output_json_path: str,
+    ):
+        export_args = [
+            "python3",
+            "-m",
+            "sharktank.examples.export_paged_llm_v1",
+            "--irpa-file",
+            irpa_path,
+            "--output-mlir",
+            output_mlir_path,
+            "--output-config",
+            output_json_path,
+        ]
+        if attention_kernel in ["decomposed", "torch_sdpa"]:
+            export_args.append("--attention-kernel")
+            export_args.append(attention_kernel)
+
+        cmd = subprocess.list2cmdline(export_args)
+        return cmd
```

Comment:
This can be separate from this PR, but we should use an in-memory API for all model exporting/importing workflows, rather than fork into a subprocess.

Users shouldn't need to chain together `python -m sharktank.examples.` and  `iree-compile ...` commands. We can aim for something like https://docs.vllm.ai/en/latest/getting_started/quickstart.html#offline-batched-inference
```python
llm = LLM(model="facebook/opt-125m")
outputs = llm.generate(prompts, sampling_params)
```

(that's as minimal as it gets - we'll want to pass options like the compilation target though)